### PR TITLE
protocol: include first new block in chain entry response

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

File: src/cryptonote_protocol/cryptonote_protocol_handler.inl [hunk 1]
```diff
@@ -1813,6 +1813,16 @@ skip:
       LOG_ERROR_CCONTEXT("Failed to handle NOTIFY_REQUEST_CHAIN.");
       return 1;
     }
+    if (r.m_block_ids.size() >= 1)
+    {
+      cryptonote::block b;
+      if (!m_core.get_block_by_hash(r.m_block_ids[1], b))
```
Comment (xiphon) [inline src/cryptonote_protocol/cryptonote_protocol_handler.inl:1819]: Should be `r.m_block_ids[0]`

Comment (moneromooo-monero) [reply src/cryptonote_protocol/cryptonote_protocol_handler.inl:1819]: No, [0] is the overlapping block, 1 is the first new block. I want this one.

Comment (xiphon) [reply src/cryptonote_protocol/cryptonote_protocol_handler.inl:1819]: Got it. Should be ` if (r.m_block_ids.size() >= 2)` then

Comment (moneromooo-monero) [reply src/cryptonote_protocol/cryptonote_protocol_handler.inl:1819]: Thanks. For reference, 7135 is what uses it (might not compile, it was full of conflicts and it's not ready so I did not bother building)

**Review by xiphon — APPROVED** [review]

**Pull request merged**